lib/delayed/job_wrapper.rb

```diff
@@ -25,7 +33,9 @@ def encode_with(coder)
     private
 
     def job
-      @job ||= ActiveJob::Base.deserialize(job_data) if job_data
+      return @job if defined?(@job)
+
+      @job = ActiveJob::Base.deserialize(job_data) if job_data
```

Comment:
Ah sweet, I like that better. Reviewing this before opening the PR I couldn't work out why I'd done it this way, probably because it was failing once I'd written the initialiser that way.